src/compiler/typescript_to_rust/azle_generate/src/candid_type/type_ref/errors/wrong_number_of_params.rs

```
@@ -13,132 +12,163 @@ use crate::{
 
 #[derive(Debug, Clone, PartialEq)]
 pub struct WrongNumberOfParams {
-    name: String,
+    name: TypeRefCandidTypes,
     param_count: usize,
     location: Location,
     modified_source: String,
     modified_range: (usize, usize),
 }
 
+#[derive(Debug, Clone, PartialEq)]
+enum TypeRefCandidTypes {
+    Record,
+    Variant,
+    Tuple,
+    Vec,
+    Opt,
+    Func,
+}
+
 impl WrongNumberOfParams {
+    fn create_annotation(&self) -> String {
+        if self.param_count == 0 {
+            "Needs to have an enclosed type here."
+        } else {
```

Let's return an unreachable error if the param_count is 1